### PR TITLE
Refactoring & Outlook.com Rules Add

### PR DESCRIPTION
**Pull request by nikitaeverywhere**
Hello! I would like to contribute to your awesome library to make it even more normalize-able!

Changes I've made:

1. Removed `is-email` dependency, as it does nothing more than checking whether e-mail matches `.*@.*\..*`, which is pretty useless for this module and can be done in a form of a simple if statement.
2. Refactored a code and an algorithm to use constants. This makes the code more extendable.
3. Added `outlook.com` provider rules.
4. Added tests for `outlook.com` provider.
5. Added `.npmignore` file which cleans unused files from being upload to npm (and downloaded to thousands of computers). Personally, I would like to also ignore `test` directory, but let it be on your own decision.

The code edits look like the whole library rewrite, but I hope everyone will benefit from this! 😄 All the original logic is preserved. I am going to link your library in my article so that's why I am proposing this changes.

Thank you!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,37 +1,49 @@
...
-  if (isNormalizeableProvider(domain)) {
-    username = username.split('+')[0]
+  var username = emailParts[0];
+  var domain = emailParts[1];
 
-    if(!/hotmail\.com$/.test(domain)) {
-      username = username.replace(/\./g, '')
+  if (normalizeableProviders.hasOwnProperty(domain)) {
+    if (normalizeableProviders[domain].hasOwnProperty('cut')) {
+      username = username.replace(normalizeableProviders[domain]['cut'], '');
```
Comment (johno) [inline index.js]:
```js
username.replace(normalizeableProviders[domain].cut, '');
```

File: index.js [hunk 2]
```diff
@@ -1,37 +1,49 @@
...
+    }
+    if (normalizeableProviders[domain].hasOwnProperty('aliasOf')) {
+      domain = normalizeableProviders[domain]['aliasOf'];
```
Comment (johno) [inline index.js]:
```js
domain = normalizeableProviders[domain].aliasOf;
```

**Review by johno — COMMENTED** [review]
Looks great! Thanks for your contribution! ❤️ 🍻 Two little nitpicks. I'm also fine ignoring the test dir as well. 👍

**Comment by nikitaeverywhere** [comment]
Ah I knew you will point to this-like props access 😄 Now fixed. You're welcome! 🍻
Looking forward to see updated package on NPM!

**Comment by johno** [comment]
🚢 Published to npm as `normalize-email@1.1.0`.

**Comment by nikitaeverywhere** [comment]
You are a rocket! 🚀 Thank you!

**Pull request merged**